### ISSUE TITLE
Trying to write log with timestamp

### ISSUE THREAD

**Issue by brunodomenici** [googleapis/nodejs-logging#800]
Hi,

I have a simple NodeJS application that reads the contents of a text file, parse it and send it to Stackdriver. I'd like to use my log source's timestamp to Stackdriver's log entry (not the timestamp that I write the log). 
I'm trying to set `timestamp` in my code, but when I do that I don't have any error but I can't find the log entry in Stackdriver.

If I do a console.log of my LogEntry I have:

```
Entry {
  metadata: {
    timestamp: '2020-04-11T06:59:59.665Z',
    resource: { type: 'global' },
    severity: 'INFO',
    insertId: '..........3pf7Ikvnp9dsty3M2TpYYE'
  },
  data: {
*** OMMITED ***
    }
  }
}
``` 

If I search for insertId in Stackdriver I can't find. However, if I don't populate `metadata.timestamp` it works just fine.

As I can see in docs, I can set the LogEntry timestamp, right? 

Any ideias?

Thanks a lot!

**Comment by laljikanjareeya** [comment]
@brunodomenici  It seems like issue in Google Cloud Console filter that you are looking for, Please refer the working demo Here:

```ts
import { Logging } from '@google-cloud/logging';

async function logEntry() {
    const insertId = '3pf7Ikvnp9dsty3M2TpYYE';
    const logName = 'my-log';
    const logging = new Logging();
    const log = logging.log(logName);

    const entry = log.entry({
        timestamp: new Date(),
        resource: { type: 'global' },
        severity: 'INFO',
        insertId
    }, {
        message: 'my test message-1'
    });

    //write log
    await log.write(entry);
    console.log(`Wrote to ${logName}`);

    //get log
    const filter = `insertId=${insertId}`;
    var options = {
        filter,
        pageSize: 10
    }
    const [entries] = await logging.getEntries(options);
    console.log('Logs:');
    entries.forEach(entry => {
        const metadata = entry.metadata;
        console.log(`timestamp: ${metadata.timestamp}`);
    });
}

logEntry();
```

**Comment by laljikanjareeya** [comment]
@brunodomenici  Where you able to solve your issue using the demo given above?

**Comment by brunodomenici** [comment]
Hi @laljikanjareeya, thank you for that! Actually, the problem wasn't in NodeJS SDK directly, I faced the same problem using Logging REST API too. I opened a support ticket but I'm not able to reproduce the issue. I think that could be related to some lag between writing and listing the log in Stackdriver. Anyways, my solution is running in production and I don't have any problem anymore...
Thanks a lot!
